packages/lesswrong/splits/useLoadedLibrary.ts

```diff
@@ -0,0 +1,43 @@
+import { useRerender } from '../components/hooks/useFirstRender';
+import { isClient } from '../lib/executionEnvironment';
+
+const libraryLoadStatus: Record<string, "loading"|"ready"> = {};
+
+export function useLoadedLibrary<T>({path, windowField}: {
+  path: string,
+  windowField: string,
+}): { ready: false } | { ready: true, library: T } {
+  const rerender = useRerender();
```

Comment:
I would prefer `const [ready, setReady] = useState(false);` (and removing this hook), because it does count as a genuine state change imo, and this would remove any possibility of infinite rerenders

Reply:
The catch is that initializing it to `false` is sometimes incorrect, since the library might already be loaded at the time this first mounts (if this isn't the first component to use the library).

Reply:
Ah I see, in that case this seems fine